File: app/src/main/java/hk/hongkongpost/app/sample/barcodereader/MainActivity.java
package hk.hongkongpost.app.sample.barcodereader;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.device.ScanManager;
import android.media.AudioAttributes;
import android.media.AudioManager;
import android.media.SoundPool;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.webkit.WebView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;


import org.jetbrains.annotations.NotNull;


import hk.hongkongpost.app.sample.barcodereader.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "MainActivity";
    private static final String LANDING_PAGE_URL = "file:///android_asset/index.html";

    private ActivityMainBinding binding;
    private WebView webView = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        webView = binding.webview;
        UROVOWebView.INSTANCE.updateUROVOSetting(webView);
        getLifecycle().addObserver(new UrovoScanner(this, new BarcodeListener() {
            @Override
            public void onScanned(@NotNull String code) {
                UROVOWebView.INSTANCE.emitBarcodeToWebView(webView, code);
            }
        }));
        webView.loadUrl(LANDING_PAGE_URL);

        binding.button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                webView.loadUrl(LANDING_PAGE_URL);
            }
        });
    }


}

File: app/src/main/java/gov/hkpost/urovo/MainActivity.kt
package gov.hkpost.urovo

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import gov.hkpost.urovo.databinding.ActivityMainBinding
import gov.hkpost.urovo.UROVOWebView.emitBarcodeToWebView
import gov.hkpost.urovo.UROVOWebView.updateUROVOSetting
import java.util.*
import kotlin.concurrent.schedule

class MainActivity : AppCompatActivity() {

    lateinit var binding: ActivityMainBinding
    private val LANDING_PAGE_URL = "file:///android_asset/index.html"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        val webView = binding.webview
        webView.updateUROVOSetting()
        webView.loadUrl(LANDING_PAGE_URL)
        binding.button.setOnClickListener {
            webView.emitBarcodeToWebView("This-is-the-barcode-received-by-urovo")
        }
    }

}